include/openPMD/Series.hpp

```diff
@@ -328,6 +331,9 @@ class SeriesImpl : public AttributableImpl
     }
 
     std::unique_ptr< ParsedInput > parseInput(std::string);
+    // use a template in order not to expose nlohmann_json to users
+    template< typename JSON >
+    void parseJsonOptions( JSON const & );
```

Comment:
Since you only call this in the `.cpp` file, you can define this helper function only in the `Series.cpp` file and in a `namespace {}` (as you already do) and avoid that it gets exposed as a function and symbol at all :)

Reply:
It's not that easy because that function accesses private members. But we can theoretically make all members of `SeriesData` public and inherit it privately in `SeriesInternal`. So, the data struct can be passed to non-friend functions for manipulation?